### ISSUE TITLE
Exposed wrong settings bricked g3 flex cameras.

### ISSUE THREAD

**Issue by incarvr6** [briis/unifiprotect#358]
### UniFi Protect HA 0.10.0

After using on g3 flex new exposed setting for high FPS 2 cameras freeze. It is still connected to network and access to local IP is available but no settings are available. After several resets ( including phisical) I managed to get back to life only one.

**Comment by incarvr6** [comment]
![IMG_20211127_230301](https://user-images.githubusercontent.com/49254419/143722779-f2c9b603-56e4-4a62-9cbb-525de807f71b.jpg)
![Screenshot_2021-11-27-21-54-54-27_40deb401b9ffe8e1df2f1cc5ba480b12](https://user-images.githubusercontent.com/49254419/143722790-cba1c73f-fb93-482a-bf06-3836511646e3.jpg)

**Comment by AngellusMortis** [comment]
Can you please update your versions of Home Assistant, UniFi Protect and the HA integration? Latest is not very helpful for troubleshooting.

**Comment by AngellusMortis** [comment]
I do not have G3 Flex cameras, so I might not be of the best help. But I assume you mean the "High FPS mode" switch inside of Home Assistant, right? This _should_ trigger the same update that UFP itself does, so it _should_ be no different than how doing it from the UFP app / WebUI is. I know with my G4 Pros, switching on/off High FPS mode that kind of "lock up" the camera for a few seconds, while it changes everything internally. But it comes back once it is done.

Something to assistant with debugging though, can you do the following and try to get me the outputted data?

1. Go to the UniFi Protect WebUI
2. Click on one of your G3 Flex cameras
3. Go to the Recording Tab. Do you have a "High Frame Rate Mode" under Recording Quality:
![image](https://user-images.githubusercontent.com/490848/143722887-91345001-a469-4670-8a1c-1ff4335dec5f.png)
4. If so turn _off_ High Frame Rate Mode
5. Copy the Camera ID out of the URL (it should look like `<protect_url>/protect/devices/####` where `####` is your Camera ID
6. Go to `<protect_url>/proxy/protect/api/cameras/####`. Copy everything on this page and paste it back in to this issue.
7. If you _do_ have an High Frame Rate Mode (question from #3), go back and turn it back on.
8. Reload the URL from #6 and post the _new_ data here.

**Comment by incarvr6** [comment]
I'm on phone right now so very hard to type and switch between multiple windows.. that's regarding versions.

No on G3 flex or G3 bullet there is no settings for high FPS

**Comment by AngellusMortis** [comment]
There is no rush on getting the info. It would be great if you can get the JSON output from 6. for me as soon as you get the chance though (when you are back at a computer?).

**Comment by incarvr6** [comment]
Yes it is still connecting to network and gui page is available , I can even login but that about it.
It is not recognised in protect and shows up as offline. Using resset button does not work either.
Now regarding 5 and 6 how exactly to do it?

**Comment by AngellusMortis** [comment]
The good news is then we _should_ be able to recover the devices. I will have to do a bunch of poking around on my cameras to figure out how to do it through. Are you able to turn the switch back off for them in Home Assistant?

5/6 you have to do from the Web browser version of Protect. You will definitely _not_ want to be a phone. So go to the IP address of your Protect instance directly. 

If you open the settings for one of the cameras, you should see the URL at the top of the screen change from 

`https://<protect_ip>/protect/devices` to `https://<protect_ip>/protect/devices/#####`

Copy the value after `devices/` and then go to 

`https://<protect_ip>/proxy/protect/api/cameras/#####`, replacing `#####` with the value you copied from above.

It should look something like this:

![image](https://user-images.githubusercontent.com/490848/143723440-5efddef6-5050-4cd3-bc4a-df9c4567fa07.png)

Copy and paste that into the issue.

**Comment by incarvr6** [comment]
`{"isDeleting":false,"mac":"18E829082458","host":"192.168.1.24","connectionHost":"192.168.1.21","type":"UVC G3 Flex","name":"Back-garden","upSince":1638055851319,"uptime":76,"lastSeen":1638055927319,"connectedSince":null,"state":"DISCONNECTED","hardwareRevision":"8","firmwareVersion":"4.46.13","latestFirmwareVersion":"4.46.13","firmwareBuild":"1380006.211025.934","isUpdating":false,"isAdopting":false,"isAdopted":true,"isAdoptedByOther":false,"isProvisioned":true,"isRebooting":false,"isSshEnabled":false,"canAdopt":false,"isAttemptingToConnect":false,"lastMotion":1638046975837,"micVolume":100,"isMicEnabled":true,"isRecording":false,"isMotionDetected":false,"isSmartDetected":false,"phyRate":100,"hdrMode":true,"videoMode":"default","isProbingForWifi":false,"apMac":null,"apRssi":null,"elementInfo":null,"chimeDuration":300,"isDark":true,"lastPrivacyZonePositionId":null,"lastRing":null,"isLiveHeatmapEnabled":false,"anonymousDeviceId":"e347b93a-87e9-5c66-87a5-62916530b42d","eventStats":{"motion":{"today":88,"average":29,"lastDays":[38,17,6,16,28,7,94],"recentHours":[0,0,1,0,2,4,2,1,17,0,0,6,3]},"smart":{"today":0,"average":0,"lastDays":[0,0,0,0,0,0,0]}},"videoReconfigurationInProgress":false,"wiredConnectionState":{"phyRate":100},"channels":[{"id":0,"videoId":"video1","name":"High","enabled":true,"isRtspEnabled":true,"rtspAlias":"2QqppTt926VNUlbc","width":1920,"height":1080,"fps":25,"bitrate":6000000,"minBitrate":32000,"maxBitrate":6000000,"minClientAdaptiveBitRate":0,"minMotionAdaptiveBitRate":750000,"fpsValues":[1,2,3,4,5,6,8,9,10,12,15,16,18,20,24,25],"idrInterval":5},{"id":1,"videoId":"video3","name":"Medium","enabled":true,"isRtspEnabled":false,"rtspAlias":null,"width":1024,"height":576,"fps":25,"bitrate":1500000,"minBitrate":32000,"maxBitrate":2000000,"minClientAdaptiveBitRate":150000,"minMotionAdaptiveBitRate":750000,"fpsValues":[1,2,3,4,5,6,8,9,10,12,15,16,18,20,24,25],"idrInterval":5},{"id":2,"videoId":"video2","name":"Low","enabled":true,"isRtspEnabled":false,"rtspAlias":null,"width":640,"height":360,"fps":15,"bitrate":300000,"minBitrate":32000,"maxBitrate":1000000,"minClientAdaptiveBitRate":0,"minMotionAdaptiveBitRate":300000,"fpsValues":[1,2,3,4,5,6,8,9,10,12,15],"idrInterval":5}],"ispSettings":{"aeMode":"auto","irLedMode":"auto","irLedLevel":255,"wdr":1,"icrSensitivity":0,"brightness":50,"contrast":50,"hue":50,"saturation":50,"sharpness":50,"denoise":50,"isFlippedVertical":true,"isFlippedHorizontal":true,"isAutoRotateEnabled":true,"isLdcEnabled":false,"is3dnrEnabled":true,"isExternalIrEnabled":false,"isAggressiveAntiFlickerEnabled":false,"isPauseMotionEnabled":false,"dZoomCenterX":50,"dZoomCenterY":50,"dZoomScale":0,"dZoomStreamId":4,"focusMode":"ztrig","focusPosition":0,"touchFocusX":1001,"touchFocusY":1001,"zoomPosition":0,"mountPosition":null},"talkbackSettings":{"typeFmt":"aac","typeIn":"serverudp","bindAddr":"0.0.0.0","bindPort":7004,"filterAddr":null,"filterPort":null,"channels":1,"samplingRate":22050,"bitsPerSample":16,"quality":100},"osdSettings":{"isNameEnabled":true,"isDateEnabled":true,"isLogoEnabled":true,"isDebugEnabled":false},"ledSettings":{"isEnabled":false,"blinkRate":0},"speakerSettings":{"isEnabled":true,"areSystemSoundsEnabled":false,"volume":80},"recordingSettings":{"prePaddingSecs":2,"postPaddingSecs":2,"minMotionEventTrigger":1000,"endMotionEventDelay":3000,"suppressIlluminationSurge":false,"mode":"always","geofencing":"off","motionAlgorithm":"enhanced","enablePirTimelapse":false,"useNewMotionAlgorithm":true},"smartDetectSettings":{"objectTypes":[]},"recordingSchedules":[],"motionZones":[{"id":1,"name":"Default","color":"#AB46BC","points":[[0,0],[1,0],[1,1],[0,1]],"sensitivity":50}],"privacyZones":[{"id":1,"name":"hass zone","color":"#85BCEC","points":[[0,0],[0,0],[0,0],[0,0]]}],"smartDetectZones":[{"id":1,"name":"Default","color":"#AB46BC","points":[[0,0],[1,0],[1,1],[0,1]],"sensitivity":50,"objectTypes":[]}],"smartDetectLines":[],"stats":{"rxBytes":3614669399,"txBytes":113732541764,"wifi":{"channel":null,"frequency":null,"linkSpeedMbps":null,"signalQuality":50,"signalStrength":0},"battery":{"percentage":null,"isCharging":false,"sleepState":"disconnected"},"video":{"recordingStart":1637096504104,"recordingEnd":1638048158885,"recordingStartLQ":1637075297584,"recordingEndLQ":1638048158868,"timelapseStart":1627264308058,"timelapseEnd":1638047774347,"timelapseStartLQ":1614889140847,"timelapseEndLQ":1638047279359},"storage":{"used":249108103168,"rate":279.953547890151},"wifiQuality":50,"wifiStrength":0},"featureFlags":{"canAdjustIrLedLevel":false,"canMagicZoom":false,"canOpticalZoom":false,"canTouchFocus":false,"hasAccelerometer":true,"hasAec":false,"hasBattery":false,"hasBluetooth":false,"hasChime":false,"hasExternalIr":false,"hasIcrSensitivity":true,"hasLdc":true,"hasLedIr":true,"hasLedStatus":true,"hasLineIn":false,"hasMic":true,"hasPrivacyMask":true,"hasRtc":false,"hasSdCard":false,"hasSpeaker":false,"hasWifi":false,"hasHdr":true,"hasAutoICROnly":true,"videoModes":["default"],"videoModeMaxFps":[],"hasMotionZones":true,"hasLcdScreen":false,"mountPositions":[],"smartDetectTypes":[],"motionAlgorithms":["enhanced"],"hasSquareEventThumbnail":true,"hasPackageCamera":false,"privacyMaskCapability":{"maxMasks":16,"rectangleOnly":false},"focus":{"steps":{"max":null,"min":null,"step":null},"degrees":{"max":null,"min":null,"step":null}},"pan":{"steps":{"max":null,"min":null,"step":null},"degrees":{"max":null,"min":null,"step":null}},"tilt":{"steps":{"max":null,"min":null,"step":null},"degrees":{"max":null,"min":null,"step":null}},"zoom":{"steps":{"max":null,"min":null,"step":null},"degrees":{"max":null,"min":null,"step":null}},"hasSmartDetect":false},"pirSettings":{"pirSensitivity":100,"pirMotionClipLength":15,"timelapseFrameInterval":15,"timelapseTransferInterval":600},"lcdMessage":{},"wifiConnectionState":{"channel":null,"frequency":null,"phyRate":null,"signalQuality":null,"signalStrength":null,"ssid":null},"lenses":[],"id":"604134a103e42103e70003ee","isConnected":false,"platform":"s2l","hasSpeaker":false,"hasWifi":false,"audioBitrate":64000,"canManage":false,"isManaged":true,"modelKey":"camera"}`

**Comment by incarvr6** [comment]
This log is from nonworking camera

Yes switch is available in HA  but none of functions works

**Comment by incarvr6** [comment]
`{"isDeleting":false,"mac":"18E829CF9F76","host":"192.168.1.22","connectionHost":"192.168.1.21","type":"UVC G3 Flex","name":"Living room","upSince":1638052272323,"uptime":4375,"lastSeen":1638056647323,"connectedSince":1638052303647,"state":"CONNECTED","hardwareRevision":"8","firmwareVersion":"4.46.13","latestFirmwareVersion":"4.46.13","firmwareBuild":"1380006.211025.934","isUpdating":false,"isAdopting":false,"isAdopted":true,"isAdoptedByOther":false,"isProvisioned":true,"isRebooting":false,"isSshEnabled":false,"canAdopt":false,"isAttemptingToConnect":false,"lastMotion":1638055707829,"micVolume":100,"isMicEnabled":true,"isRecording":true,"isMotionDetected":false,"isSmartDetected":false,"phyRate":100,"hdrMode":true,"videoMode":"default","isProbingForWifi":false,"apMac":null,"apRssi":null,"elementInfo":null,"chimeDuration":300,"isDark":false,"lastPrivacyZonePositionId":null,"lastRing":null,"isLiveHeatmapEnabled":false,"anonymousDeviceId":"1b4be45b-c315-5c92-b6ed-2faeb17ce3bc","eventStats":{"motion":{"today":149,"average":122,"lastDays":[70,104,225,184,165,40,67],"recentHours":[4,10,2,6,5,4,9,14,10,0,0,36,12]},"smart":{"today":0,"average":0,"lastDays":[0,0,0,0,0,0,0]}},"videoReconfigurationInProgress":false,"wiredConnectionState":{"phyRate":100},"channels":[{"id":0,"videoId":"video1","name":"High","enabled":true,"isRtspEnabled":true,"rtspAlias":"yhZ9cP7sSc4o1yM1","width":1920,"height":1080,"fps":25,"bitrate":6000000,"minBitrate":32000,"maxBitrate":6000000,"minClientAdaptiveBitRate":0,"minMotionAdaptiveBitRate":750000,"fpsValues":[1,2,3,4,5,6,8,9,10,12,15,16,18,20,24,25],"idrInterval":5},{"id":1,"videoId":"video3","name":"Medium","enabled":true,"isRtspEnabled":false,"rtspAlias":null,"width":1024,"height":576,"fps":25,"bitrate":833333,"minBitrate":32000,"maxBitrate":2000000,"minClientAdaptiveBitRate":150000,"minMotionAdaptiveBitRate":750000,"fpsValues":[1,2,3,4,5,6,8,9,10,12,15,16,18,20,24,25],"idrInterval":5},{"id":2,"videoId":"video2","name":"Low","enabled":true,"isRtspEnabled":false,"rtspAlias":null,"width":640,"height":360,"fps":15,"bitrate":681508,"minBitrate":32000,"maxBitrate":1000000,"minClientAdaptiveBitRate":0,"minMotionAdaptiveBitRate":681508,"fpsValues":[1,2,3,4,5,6,8,9,10,12,15],"idrInterval":5}],"ispSettings":{"aeMode":"auto","irLedMode":"auto","irLedLevel":255,"wdr":1,"icrSensitivity":0,"brightness":50,"contrast":50,"hue":50,"saturation":50,"sharpness":50,"denoise":50,"isFlippedVertical":false,"isFlippedHorizontal":false,"isAutoRotateEnabled":true,"isLdcEnabled":false,"is3dnrEnabled":true,"isExternalIrEnabled":false,"isAggressiveAntiFlickerEnabled":false,"isPauseMotionEnabled":false,"dZoomCenterX":50,"dZoomCenterY":50,"dZoomScale":0,"dZoomStreamId":4,"focusMode":"ztrig","focusPosition":0,"touchFocusX":1001,"touchFocusY":1001,"zoomPosition":0,"mountPosition":null},"talkbackSettings":{"typeFmt":"aac","typeIn":"serverudp","bindAddr":"0.0.0.0","bindPort":7004,"filterAddr":null,"filterPort":null,"channels":1,"samplingRate":22050,"bitsPerSample":16,"quality":100},"osdSettings":{"isNameEnabled":true,"isDateEnabled":true,"isLogoEnabled":true,"isDebugEnabled":false},"ledSettings":{"isEnabled":false,"blinkRate":0},"speakerSettings":{"isEnabled":true,"areSystemSoundsEnabled":false,"volume":80},"recordingSettings":{"prePaddingSecs":2,"postPaddingSecs":2,"minMotionEventTrigger":1000,"endMotionEventDelay":3000,"suppressIlluminationSurge":false,"mode":"always","geofencing":"off","motionAlgorithm":"enhanced","enablePirTimelapse":false,"useNewMotionAlgorithm":true},"smartDetectSettings":{"objectTypes":[]},"recordingSchedules":[],"motionZones":[{"id":1,"name":"Default","color":"#AB46BC","points":[[0,0],[1,0],[1,1],[0,1]],"sensitivity":50}],"privacyZones":[],"smartDetectZones":[{"id":1,"name":"Default","color":"#AB46BC","points":[[0,0],[1,0],[1,1],[0,1]],"sensitivity":50,"objectTypes":[]}],"smartDetectLines":[],"stats":{"rxBytes":113919028,"txBytes":4269087677,"wifi":{"channel":null,"frequency":null,"linkSpeedMbps":null,"signalQuality":50,"signalStrength":0},"battery":{"percentage":null,"isCharging":false,"sleepState":"disconnected"},"video":{"recordingStart":1637098043730,"recordingEnd":1638056690157,"recordingStartLQ":1637098013833,"recordingEndLQ":1638056693174,"timelapseStart":1634829165591,"timelapseEnd":1638056488138,"timelapseStartLQ":1614889127965,"timelapseEndLQ":1638056173190},"storage":{"used":204010946560,"rate":203.009378976963},"wifiQuality":50,"wifiStrength":0},"featureFlags":{"canAdjustIrLedLevel":false,"canMagicZoom":false,"canOpticalZoom":false,"canTouchFocus":false,"hasAccelerometer":true,"hasAec":false,"hasBattery":false,"hasBluetooth":false,"hasChime":false,"hasExternalIr":false,"hasIcrSensitivity":true,"hasLdc":true,"hasLedIr":true,"hasLedStatus":true,"hasLineIn":false,"hasMic":true,"hasPrivacyMask":true,"hasRtc":false,"hasSdCard":false,"hasSpeaker":false,"hasWifi":false,"hasHdr":true,"hasAutoICROnly":true,"videoModes":["default"],"videoModeMaxFps":[],"hasMotionZones":true,"hasLcdScreen":false,"mountPositions":[],"smartDetectTypes":[],"motionAlgorithms":["enhanced"],"hasSquareEventThumbnail":true,"hasPackageCamera":false,"privacyMaskCapability":{"maxMasks":16,"rectangleOnly":false},"focus":{"steps":{"max":null,"min":null,"step":null},"degrees":{"max":null,"min":null,"step":null}},"pan":{"steps":{"max":null,"min":null,"step":null},"degrees":{"max":null,"min":null,"step":null}},"tilt":{"steps":{"max":null,"min":null,"step":null},"degrees":{"max":null,"min":null,"step":null}},"zoom":{"steps":{"max":null,"min":null,"step":null},"degrees":{"max":null,"min":null,"step":null}},"hasSmartDetect":false},"pirSettings":{"pirSensitivity":100,"pirMotionClipLength":15,"timelapseFrameInterval":15,"timelapseTransferInterval":600},"lcdMessage":{},"wifiConnectionState":{"channel":null,"frequency":null,"phyRate":null,"signalQuality":null,"signalStrength":null,"ssid":null},"lenses":[],"id":"604134a200112103e70003ef","isConnected":true,"platform":"s2l","hasSpeaker":false,"hasWifi":false,"audioBitrate":64000,"canManage":false,"isManaged":true,"modelKey":"camera"}`

This is from working one

**Comment by AngellusMortis** [comment]
Okay. I am still digging into the file system of the camera to figure out how to manually disable the video mode, but while I do that, you need to enable SSH for any Camera that is bricked. I apologize ahead of time if this is too much coding for you.

Everything below is shown in a "Chrome based" browser like Chrome, Microsoft Edge, Opera, Brave, etc.

1. Again go to the Protect Web Application. 
2. Open your developer tools (Windows: CTRL+SHIFT+I should work, you should also be able to find it from the menus somewhere)
3. Click on the network tab. Reload the page
4. You should see something like this: 
![image](https://user-images.githubusercontent.com/490848/143723915-1492020d-6aec-40f3-a4fc-79779dabdc6e.png)
5. Click the first item in the list of requests.
6. Expand "Response Headers" if it is not already expanded and copy out the value for "X-CSRF-Token", in my case, it is `003d8a9e-170b-4631-91c6-52a61efd1ad5`
![image](https://user-images.githubusercontent.com/490848/143723939-2ee47ab6-4537-4e1b-8301-8bf8714d72c1.png)
7. Run the following Javascript in your console tab for _each_ camera that is broken. Replace `AAAAA` with the ID for the camera (just like you got it from before above) and replace `CCCCC` with your CSRF token from 6.

```javascript
function enableSSH(id, csrf) {
    var url = "/proxy/protect/api/cameras/" + id;

    var xhr = new XMLHttpRequest();
    xhr.open("PATCH", url);

    xhr.setRequestHeader("Accept", "application/json");
    xhr.setRequestHeader("Content-Type", "application/json");
    xhr.setRequestHeader("X-CSRF-Token", csrf);

    xhr.onreadystatechange = function () {
    if (xhr.readyState === 4) {
        console.log(xhr.status);
        console.log(xhr.responseText);
    }};

    var data = `{
    "isSshEnabled": true
    }`;

    xhr.send(data);
}

enableSSH("AAAAA", "CCCCC")
```

The whole thing should look something like this:

![image](https://user-images.githubusercontent.com/490848/143723976-23467724-4fea-41de-9648-82b9076292b9.png)


Next one will be on how to SSH into the cameras and verify it is working.

**Comment by AngellusMortis** [comment]
If you are on Windows, make sure OpenSSH is installed: https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_install_firstuse. You should not need to do anything if you are on a Mac (I do not think, otherwise search "enable SSH on MacOS" online).

Go to your Protect settings (URL should look like `https://<protect_ip>/protect/settings`). Find your device password, reveal it and copy it out. Put it somewhere for later.

![image](https://user-images.githubusercontent.com/490848/143724037-4159f14d-0a47-4e0a-a3f3-ad6fa451ee8e.png)

Open up a terminal. On Windows Command Prompt or PowerShell will work.

Run the command:

```bash
ssh ubnt@<ip_address_of_camera`
```

Response yes if prompted and then enter the password when prompted.


If everything worked, it should look something like this:

![image](https://user-images.githubusercontent.com/490848/143724072-22f83912-5085-490d-a45c-4655d5c6da6a.png)

Now I am going to keep digging to figure out how to manually reset the video mode for you.

**Comment by incarvr6** [comment]
ok but issue is not just missing video feed it is missing all settings , and it is not recognized in protect.

**Comment by AngellusMortis** [comment]
From the JSON you posted above, it looks like they are still there, it is just the camera is "disconnected" so Protect will not let you do anything in the UI. The Javascript XHR request I posted _might_ still work.

**Comment by incarvr6** [comment]
well nothing is no where near to your screenshots on my win pc

**Comment by AngellusMortis** [comment]
You can also try this troubleshooting guide from Ubiquiti to enable SSH for all of your devices:

https://help.ui.com/hc/en-us/articles/1500007460721-UniFi-Protect-Advanced-troubleshooting-methods#ssh

Though, that looks like it is only for the UDM Pro. If you have UNVR, that may not work.

**Comment by AngellusMortis** [comment]
Anyways, figured it out. After you get SSH enabled for the bricked cameras, SSH into each of them and then run

```bash
sed -i 's/"videoMode": "highFps"/"videoMode": "default"/' /var/etc/persistent/ubnt_encoder.conf
```

That should fix the issue. For my G4 Instant, it would basically boot, try to start, fail, and then reboot. I had to edit the file _quickly_ before it tried to start the video encoder on the device. So it took me like 3 or 4 tries before I got it working.

**Comment by incarvr6** [comment]
I still have not enabled SSH on camera
I found video with more easier instructions but I can't find right buttons to change text
![Uploading IMG20211128013053.jpg…]()

**Comment by incarvr6** [comment]
I mean save

**Comment by AngellusMortis** [comment]
If you are using vi/vim, it is `:wq`. If you are using nano, it is CTRL+O and CTRL+X.

**Comment by incarvr6** [comment]
And safe?

**Comment by AngellusMortis** [comment]
That will do both. For vi/vim, `:` means input command. Then `w` means write and `q` means quit (so save and quit). For nano, CTRL+O is Overwrite and CTRL+X is exit (so save and quit).

**Comment by AngellusMortis** [comment]
Also, just to verify you are not going through all of this for nothing, but you have tried to forget the device and then hard reset it with the reset button _on_ the device right? That did not work?

**Comment by incarvr6** [comment]
Yes reset  with button did not help.
Sorry to be so dumb I still cant understand commands if I press ctrl X nothing happens

**Comment by AngellusMortis** [comment]
You only use CTRL+X if you are using nano. How did you open the file? What video are you using (so I can follow along and try to help)?

**Comment by incarvr6** [comment]
https://www.youtube.com/watch?v=UVJA6i22XAM

**Comment by AngellusMortis** [comment]
Ugh. That video has vi _and_ nano in it. What command are you running to edit the file?

**Comment by incarvr6** [comment]
vi /usr/share/unifi-protect/app/config/config.json

**Comment by AngellusMortis** [comment]
Then this should be the full sets to edit the file:

1. Run `vi /usr/share/unifi-protect/app/config/config.json`
2. Press the down arrow until you get to the `enableSsh` line.
3. Press -> until you get your cursor right before the `f`:
![image](https://user-images.githubusercontent.com/490848/143725934-a301a776-5da3-4d30-8f4e-aea705b259db.png)
4. Type `i` followed by `Del` until `false` is gone
![image](https://user-images.githubusercontent.com/490848/143725944-b2e1c638-0c63-43e9-9a85-d1e4f49ed159.png)
5. Type `true`
6. Double tap `Esc`, `INSERT` should go away from the bottom of the screen
7. Type `:wq`.

The file should save and quit. If you open it again, you should see `enableSsh: true` now.

**Comment by incarvr6** [comment]
:wq dose nothing

**Comment by AngellusMortis** [comment]
Oh, did you press enter afterwards? I kind of forgot to put that.

**Comment by incarvr6** [comment]
I'm so far and if I press anything I get windows error sound
![16380658340664203862758339857056](https://user-images.githubusercontent.com/49254419/143726287-419b9577-7db3-4a66-b0a7-2ee7090dccfb.jpg)

**Comment by incarvr6** [comment]
Or after pressing :wq I get recording @w on bottom

**Comment by AngellusMortis** [comment]
I am not able to reproduce the issue with PowerShell on my computer. I do get the Windows ding when I double tap Esc, but :wq works just fine.

You could also try this from a new PowerShell instance:

```bash
scp root@192.168.51.16:/usr/share/unifi-protect/app/config/config.json ./Desktop
```

Then that will copy the file to your desktop. Then you can open the file in Notepad and edit like normal. 

After you are all done, run the following command to put it back.

```bash
scp ./Desktop/config.json root@192.168.51.16:/usr/share/unifi-protect/app/config/config.json 
```

**Comment by incarvr6** [comment]
You must be thinking that im a moron by now :)

I still cant make it work, it downloads somewhare  but not on desktop

**Comment by incarvr6** [comment]
Right finaly i managed to enable ssh on cameras. I had to use cmd instead of powershell. Now i can reach all other cameras  but not the one that is not working.

**Comment by incarvr6** [comment]
![IMG20211128031720](https://user-images.githubusercontent.com/49254419/143727484-8011d159-8217-40ad-9f08-772323dac436.jpg)

It is on network

**Comment by AngellusMortis** [comment]
The one that is not working is probably not being provisioned by UFP. 

At this point I am out of ideas. I recommend trying the UI forums, Discord or connect Ubiquiti directly. Maybe try resetting it a few more times. I really wish I could help more.

**Comment by incarvr6** [comment]
It seems that it is stuck in some boot loop , connects to network  and stays like that. strange that gui is accssesable but not ssh.
Is there any soft resset command?

**Comment by AngellusMortis** [comment]
You are able to get to the WebUI for the camera directly? Can you get to http://<camera_ip>/camera/system?

There is a "Reset To Defaults" link there you might try. 

If that does not work, you can use an old firmware to downgrade the camera. They hopefully will unbrick it so you can readopt it and upgrade it again.

![image](https://user-images.githubusercontent.com/490848/143728043-21e1e980-d2ef-4956-9996-eedc22c0d862.png)

https://www.ui.com/download/unifi-protect/unifi-video-g3-flex-camera/uvc-g3-flex/uvc-g3-firmware-4300

**Comment by incarvr6** [comment]
no no settings are available there

**Comment by incarvr6** [comment]
Well after about 10 + hard resets  I managed to bring it back to life. Weirdly I didn't have to re add camera to protect, all settings  stayed same.

**Comment by AngellusMortis** [comment]
I am glad you finally got it working. I am going to close this out for now then. We pushed out a version update to temporarily disable the High FPS switch, but the issue should be resolved in 0.11 with the switch re-enabled.
